### PR TITLE
Cleanup grind and add serving capability via dhttpd

### PR DESCRIPTION
**Pull request by jcollins-g**
This is the beginning of a short series of dartdoc infrastructure PRs to increase release reliability and velocity.

TL;DR:

- Centralize grind script process launching
- Port flutter bot to a grind target
- Add web serving targets `serve-sdk-docs` and `serve-flutter-docs`.
- Drop duplicitous link checking -- this is now part of dartdoc proper.

--- 

Many times I find it useful to have local serving versions of docs for browsing.  To make this easier for other developers I've written some parts of what I do into the grind script.  Two new targets, `serve-sdk-docs` and `serve-flutter-docs` build and launch a Dart web server for browsing documentation for the SDK and flutter.

This also cleans up the grind script so that we launch processes via a single entry point (`_SubprocessLauncher`), which logs the command being executed as well as any pertinent environment variables, the working directory, etc in a manner familiar to shell script authors but hopefully readable to others as well.  This used to be done inconsistently making for challenging debugging.

The refactor also makes it a little safer to do these tasks in parallel -- this is why there is the creation of throwaway pub caches and safer temporary directory creation.  A followup will have a "serve-everything" grind target that will parallelize dartdoc to save time and serve the SDK, the test package, and the flutter docs on three different ports from a single command.

Other followups will include patching new versions of dartdoc into the SDK (temporarily for a test, and for release) and Flutter automatically, with generation of review docs.

### CODE REVIEW

File: tool/grind.dart [hunk 1]
```diff
@@ -3,24 +3,120 @@
...
-import 'package:html/dom.dart';
-import 'package:html/parser.dart' show parse;
 import 'package:path/path.dart' as path;
 import 'package:yaml/yaml.dart' as yaml;
 
 main([List<String> args]) => grind(args);
 
-final Directory docsDir =
-    new Directory(path.join('${Directory.systemTemp.path}', defaultOutDir));
+final Directory dartdocDocsDir = Directory.systemTemp.createTempSync('dartdoc');
```
Comment (devoncarew) [inline tool/grind.dart]: This script creates these temp dirs on every run, whether they're used or not. They're in temp, so it may not actually matter, but consider not creating them until the variables are first referenced?

File: tool/grind.dart [hunk 2]
```diff
@@ -3,24 +3,120 @@
...
+  }
+
+  /// A wrapper around start/await process.exitCode that will display the
+  /// output of the executable continuously and fail on non-zero exit codes.
+  /// Makes running programs in grinder similar to set -ex for bash, even on
+  /// Windows (though some of the bashisms will no longer make sense).
+  /// TODO(jcollins-g): move this to grinder?
+  Future runStreamed(String executable, List<String> arguments,
+      {String workingDirectory}) async {
+    stderr.write('$prefix+ ');
```
Comment (devoncarew) [inline tool/grind.dart]: you want stderr here?

File: tool/travis.sh [hunk 3]
```diff
@@ -22,25 +22,7 @@ if [ "$DARTDOC_BOT" = "sdk-docs" ]; then
...
-  # Build the flutter docs.
-  cd doc/flutter
-  ./bin/flutter --version
-  ./bin/flutter precache
-  ( cd  dev/tools; pub get )
-  ./bin/cache/dart-sdk/bin/dart dev/tools/dartdoc.dart
-
-  # The above script validates the generation; we echo the main file here.
-  cat dev/docs/doc/index.html
+  pub run grinder build-flutter-docs
```
Comment (devoncarew) [inline tool/travis.sh]: 👍 awesome - nice to make it easy for other people to run this

File: tool/grind.dart (hunk 1 above)
Comment (jcollins-g) [reply tool/grind.dart]: Done.

File: tool/grind.dart (hunk 2 above)
Comment (jcollins-g) [reply tool/grind.dart]: Yes.  Modeled after set -x in bash, which echos the commands used to stderr.
